File: java8/src/test/java/cucumber/runtime/java8/test/ParameterTypes.java
package cucumber.runtime.java8.test;

import cucumber.api.Configuration;
import cucumber.api.TypeRegistry;
import io.cucumber.datatable.DataTableType;
import cucumber.runtime.java8.test.LambdaStepdefs.Person;

import java.util.Map;

import static java.util.Locale.ENGLISH;

public class ParameterTypes implements Configuration {

    @Override
    public TypeRegistry createTypeRegistry() {
        final TypeRegistry typeRegistry = new TypeRegistry(ENGLISH);

        typeRegistry.defineDataTableType(new DataTableType(
            "person",
            Person.class,
            (Map<String, String> map) -> {
                Person person = new Person();
                person.first = map.get("first");
                person.last = map.get("last");
                return person;
            }));


        return typeRegistry;
    }
}

File: kotlin-java8/src/test/kotlin/cucumber/runtime/kotlin/test/ParameterTypes.kt
package cucumber.runtime.kotlin.test

import cucumber.api.Configuration
import cucumber.api.TypeRegistry
import io.cucumber.datatable.DataTableType
import io.cucumber.datatable.TableEntryTransformer
import java.util.Locale.ENGLISH

class ParameterTypes : Configuration {

    override fun createTypeRegistry(): TypeRegistry {
        val typeRegistry = TypeRegistry(ENGLISH)

        typeRegistry.defineDataTableType(DataTableType(
                LambdaStepdefs.Person::class.java,
                TableEntryTransformer<LambdaStepdefs.Person>
                { map: Map<String, String> ->
                    val person = LambdaStepdefs.Person()
                    person.first = map.get("first")
                    person.last = map.get("last")
                    person
                }))


        return typeRegistry
    }
}
